operate/qa/integration-tests/src/test/java/io/camunda/operate/schema/SchemaManagerIT.java

```diff
@@ -0,0 +1,178 @@
+    createIndex(idxName("index-2.3.4_"), List.of(Map.of("test_name2", "test_value2")));
+    createIndex(idxName("index-3.4.5_"), List.of(Map.of("test_name3", "test_value3")));
+
+    // set reindex settings
+    schemaManager.setIndexSettingsFor(
+        Map.of(
+            NUMBERS_OF_REPLICA, NO_REPLICA,
+            REFRESH_INTERVAL, NO_REFRESH),
+        idxName("index-1.2.3_"));
+    schemaManager.setIndexSettingsFor(
+        Map.of(REFRESH_INTERVAL, NO_REFRESH), idxName("index-2.3.4_"));
+    schemaManager.setIndexSettingsFor(
+        Map.of(NUMBERS_OF_REPLICA, "5", REFRESH_INTERVAL, NO_REFRESH), idxName("index-3.4.5_"));
+
+    // update number of replicas for each index
+    schemaManager.checkAndUpdateIndices();
+
+    final Map<String, String> reindexSettings1 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-1.2.3_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings1)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+
+    final Map<String, String> reindexSettings2 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-2.3.4_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings2)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+
+    final Map<String, String> reindexSettings3 =
+        schemaManager.getIndexSettingsFor(
+            idxName("index-3.4.5_"), NUMBERS_OF_REPLICA, REFRESH_INTERVAL);
+    assertThat(reindexSettings3)
+        .containsEntry(
+            NUMBERS_OF_REPLICA,
+            DatabaseInfo.isOpensearch() ? null : String.valueOf(expectedNumberOfReplicas))
+        .containsEntry(REFRESH_INTERVAL, DatabaseInfo.isOpensearch() ? null : NO_REFRESH);
+    // Migrator uses this
+    assertThat(schemaManager.getOrDefaultNumbersOfReplica(idxName("index-1.2.3_"), "5"))
+        .isEqualTo("5");
+    assertThat(schemaManager.getOrDefaultRefreshInterval(idxName("index-1.2.3_"), "2"))
+        .isEqualTo("2");
+  }
+
+  private void mockTenantResponse() {
+    doReturn(TenantService.AuthenticatedTenants.allTenants())
+        .when(tenantService)
+        .getAuthenticatedTenants();
+  }
+
+  private void createIndex(final String indexName, final List<Map<String, String>> documents)
+      throws Exception {
+    if (DatabaseInfo.isElasticsearch()) {
+      final Map<String, ?> mapping =
+          Map.of("properties", Map.of("test_name", Map.of("type", "keyword")));
+      searchRepository.createIndex(indexName, mapping);
+      assertThat(schemaManager.getIndexNames(idxName("index*"))).contains(indexName);
```

Comment:
Not sure I understand, we don't have to do anything similar on opensearch, just elasticsearch? And what circumstances would cause index names to be created in a way to fail this assert statement, I thought we were in control of all the index names in the test